snakemake_executor_plugin_slurm/__init__.py

```diff
@@ -730,10 +785,137 @@ def check_slurm_extra(self, job):
         jobname = re.compile(r"--job-name[=?|\s+]|-J\s?")
         if re.search(jobname, job.resources.slurm_extra):
             raise WorkflowError(
-                "The --job-name option is not allowed in the 'slurm_extra' "
-                "parameter. The job name is set by snakemake and must not be "
-                "overwritten. It is internally used to check the stati of the "
-                "all submitted jobs by this workflow."
+                "The --job-name option is not allowed in the 'slurm_extra' parameter. "
+                "The job name is set by snakemake and must not be overwritten. "
+                "It is internally used to check the stati of the all submitted jobs "
+                "by this workflow."
                 "Please consult the documentation if you are unsure how to "
                 "query the status of your jobs."
             )
+
+    def create_efficiency_report(self):
```

Comment:
Maybe move this into a separate module, in order to keep init smaller?

Reply:
yes, this was the idea, initially. Yet, not possible with the `atexit` registration. Now, a refactoring might be possible. Need to look into it.